crates/cast/src/starknet_commands/account/deploy.rs

```diff
@@ -30,9 +35,143 @@ pub struct Deploy {
+async fn deploy_from_keystore(
+    provider: &JsonRpcClient<HttpTransport>,
+    chain_id: FieldElement,
+    max_fee: FieldElement,
+    wait: bool,
+    keystore_path: Utf8PathBuf,
+    account_path: Utf8PathBuf,
+) -> Result<InvokeResponse> {
+    let contents =
+        std::fs::read_to_string(account_path.clone()).context("Couldn't read account file")?;
+    let mut items: Map<String, serde_json::Value> = serde_json::from_str(&contents)
+        .map_err(|_| anyhow!("Failed to parse account file at {account_path}"))?;
+
+    if items["deployment"].is_null() {
+        bail!("No deployment field in account JSON file");
+    }
+
+    let status = items
+        .get("deployment")
+        .and_then(|deployment| deployment.get("status"))
+        .and_then(serde_json::Value::as_str)
+        .ok_or_else(|| anyhow::anyhow!("Failed to get status from account JSON file"))?;
+
+    if status == "deployed" {
+        bail!("Account already deployed");
+    }
+
+    let salt = FieldElement::from_hex_be(
+        items
+            .get("deployment")
+            .and_then(|deployment| deployment.get("salt"))
+            .and_then(serde_json::Value::as_str)
+            .ok_or_else(|| anyhow::anyhow!("Failed to get salt from account JSON file"))?,
+    )?;
+    let oz_class_hash = FieldElement::from_hex_be(
+        items
+            .get("deployment")
+            .and_then(|deployment| deployment.get("class_hash"))
+            .and_then(serde_json::Value::as_str)
+            .ok_or_else(|| anyhow::anyhow!("Failed to get salt from account JSON file"))?,
+    )?;
+
+    let private_key = SigningKey::from_keystore(keystore_path, get_keystore_password()?.as_str())?;
+    let public_key: FieldElement = {
+        let pk = items
+            .get("variant")
+            .and_then(|v| v.get("public_key"))
+            .and_then(serde_json::Value::as_str)
+            .ok_or_else(|| anyhow::anyhow!("No public_key in account JSON file"))?;
+        parse_number(pk)?
+    };
+    if public_key != private_key.verifying_key().scalar() {
+        bail!("Public key and private key from keystore do not match");
+    }
+
+    items["deployment"]["status"] = serde_json::Value::from("deployed");
+    items.get_mut("deployment").and_then(|deployment| {
+        deployment
+            .as_object_mut()
+            .expect("should be an object")
+            .remove("salt")
+    });
+    items["deployment"]["address"] = format!(
+        "{:#x}",
+        get_contract_address(
+            salt,
+            oz_class_hash,
+            &[private_key.verifying_key().scalar()],
+            FieldElement::ZERO,
+        )
+    )
+    .into();
+
+    let result = deploy_oz_account(
+        provider,
+        oz_class_hash,
+        private_key,
+        salt,
+        chain_id,
+        max_fee,
+        wait,
+    )
+    .await?;
+
+    std::fs::write(&account_path, serde_json::to_string_pretty(&items).unwrap())
+        .context("Couldn't write to account file")?;
```

Comment:
what are possible scenarios here? file already exists? path is a directory? something else? 
We should get ready for at least some of the scenarios, and prevent the once we can prevent. it would be a shame if users lots some funds, because we failed to write necessary info to a file. And my understanding is that it could happen, since we removed salt and it is required for the deployment. right?

Reply:
The salt has been removed and address added, yes.
If the write fails here though, the file wouldn't be changed though, would it?

Reply:
yup, but since we already deployed an account, doesn't that mean we loose all the information about deployed account, because it was not written to a file? and the user cannot re-deploy it, because it already exists? and in order to deploy, user had to prefund, right? so one kind of looses money

Reply:
The account file does not change, so it's still a valid file for account deployment - but you are right, the deployment would fail because it's already deployed.
What we could do is maybe check if the account is already deployed, and if so just adjust the file without sending a second tx.